### PR TITLE
Addded custom CA logic

### PR DESCRIPTION
**Pull request by k-s-dean**
Adds logic for loading in a custom CA if the certificates have been self-signed.

### CODE REVIEW

File: roles/vault/README.md [hunk 1]
```diff
@@ -40,6 +41,18 @@ Example playbook (used with OpenStack Kayobe)
...
+      copy:
+        content: "{{ external_ca_cert }}"
+        dest: "{{ '/etc/pki/ca-trust/source/anchors/rootCA.crt' if ansible_os_family == 'RedHat' else '/usr/local/share/ca-certificates/rootCA.crt' }}"
+        mode: 0600
+        no_log: True
+      become: true
+
+    - name: update system CA
+      become: true
+      shell: "{{ 'update-ca-trust' if ansible_os_family == 'RedHat' else 'update-ca-certificates' }}"
```
Comment (markgoddard) [inline roles/vault/README.md:54]: Does it need to be in the system trust store?

File: roles/vault/tasks/vault.yml [hunk 2]
```diff
@@ -16,6 +16,14 @@
       server
   become: True
 
+- name: Copy CA into vault
+  command: "docker cp {{ '/etc/pki/ca-trust/source/anchors/rootCA.crt' if ansible_os_family == 'RedHat' else '/usr/local/share/ca-certificates/rootCA.crt' }} {{ vault_docker_name }}:/usr/local/share/ca-certificates/rootCA.crt"
+  when: copy_ca == true
```
Comment (markgoddard) [inline roles/vault/tasks/vault.yml:21]: `when: copy_ca | bool`

File: roles/vault/README.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ Role variables
     * `vault_config_dir`: Directory into which to bind mount Vault configuration
   * Optional
     * `consul_container.etc_hosts`: Dict; `{<hostname>:<ip_address>}` to be added to container /etc/host
+    * `copy_ca`: bool; copy a custom CA into the vault container
```
Comment (markgoddard) [inline roles/vault/README.md:28]: Lacks a default value.

Needs a prefix, like other variables.

**Review by markgoddard — CHANGES_REQUESTED** [review]
Why does the Vault container need the CA file?

File: roles/vault/README.md (hunk 1 above)
Comment (k-s-dean) [reply roles/vault/README.md:54]: Yes, the URI module in vault.yml can't talk to vault if you use self-signed certificates.

File: roles/vault/README.md [hunk 4]
```diff
@@ -40,6 +41,18 @@ Example playbook (used with OpenStack Kayobe)
   gather_facts: True
   hosts: consul
   tasks:
+    - name: Copy rootCA
+      copy:
+        content: "{{ external_ca_cert }}"
+        dest: "{{ '/etc/pki/ca-trust/source/anchors/rootCA.crt' if ansible_os_family == 'RedHat' else '/usr/local/share/ca-certificates/rootCA.crt' }}"
```
[the remark this answers is not in the document]
Comment (k-s-dean) [reply roles/vault/README.md:47]: custom_ca.crt ?

[the remark this answers is not in the document]
Comment (markgoddard) [reply roles/vault/README.md:47]: vaultCA.crt?

**Pull request closed without merging**